includes/namespace.php

```diff
@@ -29,63 +29,112 @@
-					echo wp_kses(
-						sprintf(
-						/* translators: %s: build commands. */
-							__( 'You appear to be running an incomplete version of the plugin. Please run %s to finish installation.', 'web-stories' ),
-							'<code>composer install &amp;&amp; npm install &amp;&amp; npm run build</code>'
-						),
-						[
-							'code' => [],
-						]
-					);
-				?>
-			</p>
-		</div>
-		<?php
-	}
+// Load Compatibility class the old fashioned way.
+if ( ! class_exists( '\Google\Web_Stories\Compatibility' ) ) {
+	require_once WEBSTORIES_PLUGIN_DIR_PATH . 'includes/Compatibility.php';
+}
+
+global $web_stories_compatibility;
+
+$web_stories_error = new WP_Error();
+$extensions        = [
+	'date'   => [
+		'classes' => [
+			'DateTimeImmutable',
+		],
+	],
+	'dom'    => [
+		'classes' => [
+			'DOMAttr',
+			'DOMComment',
+			'DOMDocument',
+			'DOMElement',
+			'DOMNode',
+			'DOMNodeList',
+			'DOMText',
+			'DOMXPath',
+		],
+	],
+	'json'   => [
+		'functions' => [
+			'json_decode',
+			'json_encode',
+		],
+	],
+	'libxml' => [
+		'functions' => [
+			'libxml_use_internal_errors',
+		],
+	],
+	'spl'    => [
+		'functions' => [
+			'spl_autoload_register',
+		],
+	],
+];
 
-	add_action( 'admin_notices', __NAMESPACE__ . '\_print_missing_build_admin_notice' );
+$web_stories_compatibility = new Compatibility( $web_stories_error );
+$web_stories_compatibility->set_extensions( $extensions );
+$web_stories_compatibility->set_php_version( WEBSTORIES_MINIMUM_PHP_VERSION );
+$web_stories_compatibility->set_wp_version( WEBSTORIES_MINIMUM_WP_VERSION );
+$web_stories_compatibility->set_required_files(
+	[
+		WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/edit-story.js',
+		WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/stories-dashboard.js',
+		WEBSTORIES_PLUGIN_DIR_PATH . '/assets/js/web-stories-embed-block.js',
+	]
+);
+$web_stories_compatibility->set_class_name( '\Google\Web_Stories\Plugin' );
+
+/**
+ * Displays an admin notice about why the plugin is unable to load.
+ *
+ * @since 1.0.0
+ *
+ * @return void
+ */
+function _print_missing_build_admin_notice() {
+	global $web_stories_compatibility;
+
+	$web_stories_compatibility->run_checks();
+	$_error = $web_stories_compatibility->get_error();
+	if ( ! $_error->errors ) {
+		return;
+	}
+	?>
+	<div class="notice notice-error">
+		<p><strong><?php esc_html_e( 'Web Stories plugin could not be initialized.', 'web-stories' ); ?></strong></p>
+		<ul>
+			<?php
+			foreach ( array_keys( $_error->errors ) as $error_code ) {
+				foreach ( $_error->get_error_messages( $error_code ) as $message ) {
+					printf( '<li>%s</li>', wp_kses( $message, [ 'code' => [] ] ) );
+				}
+			}
+			?>
+		</ul>
+	</div>
+	<?php
 }
 
-if ( ! class_exists( '\Google\Web_Stories\Plugin' ) ) {
+add_action( 'admin_notices', __NAMESPACE__ . '\_print_missing_build_admin_notice' );
+
+if ( ( defined( 'WP_CLI' ) && WP_CLI ) || 'true' === getenv( 'CI' ) || 'cli' === PHP_SAPI ) {
 	// In CLI context, existence of the JS files is not required.
```

Comment:
Let's ditch this.

While existence of the JS files is not required in CLI context, this should be surfaced to the user nonetheless, because it still means the plugin is incomplete and not working as expected.

That means we can use the same checks in CLI context and regular context.

That means we can just use one `run_checks()` method in both places.

Reply:
@swissspidy I remember why we did this in the first place. Without the javascript files the unit tests fail. Building javascript will take time and massively slow down the phpunit tests runs. I have a workaround here https://github.com/google/web-stories-wp/pull/4930/commits/a43bae9b754e378f00c20b105ef5f1a9570d065d

Reply:
https://github.com/google/web-stories-wp/pull/4930/files#diff-e38e7938d6df1990969cd7d147a8d590b359fa3a96a33802c85aa5723abf0223R124